org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacCodeAssistProvider.java

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.jdt.core.dom;
+
+import java.util.Map;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.CompletionRequestor;
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.ITypeRoot;
+import org.eclipse.jdt.core.WorkingCopyOwner;
+import org.eclipse.jdt.internal.codeassist.CompletionEngine;
+import org.eclipse.jdt.internal.codeassist.ICodeAssistProvider;
+import org.eclipse.jdt.internal.core.SearchableEnvironment;
+import org.eclipse.jdt.internal.javac.completion.DOMCompletionEngine;
+
+public class JavacCodeAssistProvider implements ICodeAssistProvider {
```

Comment:
I would stick to `DOM` in the name to explicit that we don't want Javac APIs to be used in here.
Maybe later we'd rewrite a whole completion engine with Javac APIs directly, and that one would be the JavacCodeAssistProvider/JavacCompletionEngine.